src/pages/UserDashboard.tsx

```diff
@@ -12,151 +21,319 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
-            </Link>
-          </Button>
-          {user.is_superuser && (
+          <div className="flex flex-col sm:flex-row gap-3">
+            {user.is_superuser && (
+              <Button
+                variant="outline"
+                size="sm"
+                className="w-full sm:w-auto"
+                asChild
+              >
+                <Link
+                  href="/admin/questionnaire"
+                  className="gap-2 text-inherit flex items-center"
+                >
+                  <User2Icon className="h-4 w-4" />
+                  {t("admin_dashboard")}
+                </Link>
+              </Button>
+            )}
             <Button
               variant="outline"
               size="sm"
-              className="w-full sm:w-auto"
+              className="w-full sm:w-auto hidden sm:flex"
               asChild
             >
               <Link
-                href="/admin/questionnaire"
+                href={`/users/${user.username}`}
                 className="gap-2 text-inherit flex items-center"
               >
-                <User2Icon className="h-4 w-4" />
-                {t("admin_dashboard")}
+                <SquarePen className="h-4 w-4" />
+                {t("edit_profile")}
               </Link>
             </Button>
-          )}
-          <Button
-            variant="outline"
-            size="sm"
-            className="w-full sm:w-auto"
-            onClick={signOut}
-          >
-            <LogOut className="h-4 w-4" />
-            {t("sign_out")}
-          </Button>
+            <DropdownMenu>
+              <DropdownMenuTrigger asChild>
+                <Button
+                  variant="outline"
+                  size="sm"
+                  className="px-2 w-full sm:w-auto"
+                >
+                  <CareIcon icon="l-ellipsis-v" className="text-inherit" />
+                </Button>
+              </DropdownMenuTrigger>
+              <DropdownMenuContent align="end" className="w-40">
+                {user.is_superuser && (
+                  <DropdownMenuItem className="cursor-pointer flex items-center gap-2 text-xs w-full sm:w-auto">
+                    <Link
+                      href="/admin/questionnaire"
+                      className="flex items-center gap-2 w-full text-inherit"
+                    >
+                      <User2Icon className="h-4 w-4" />
+                      {t("admin_dashboard")}
+                    </Link>
+                  </DropdownMenuItem>
+                )}
+                <DropdownMenuItem
+                  className="cursor-pointer flex items-center gap-2 text-xs w-full sm:w-auto"
+                  onClick={signOut}
+                >
+                  <LogOut className="h-4 w-4" />
+                  {t("sign_out")}
+                </DropdownMenuItem>
+                <DropdownMenuItem className="cursor-pointer flex sm:hidden items-center gap-2 text-xs w-full sm:w-auto">
```

Comment:
let's move edit profile above sign out (don't forget to adjust the visibility attribute once you remove button outside of the dropdown menu).